var/spack/repos/builtin/packages/dyninst/package.py

```diff
@@ -34,59 +34,33 @@ class Dyninst(CMakePackage):
     version("10.2.1", sha256="8077c6c7a12577d2ffdcd07521c1eb1b7367da94d9a7ef10bf14053aeaae7ba1")
     version("10.2.0", sha256="4212b93bef4563c7de7dce4258e899bcde52315a571087e87fde9f8040123b43")
     version("10.1.0", sha256="4a121d70c1bb020408a7a697d74602e18250c3c85800f230566fcccd593c0129")
-    version("10.0.0", sha256="542fccf5c57c4fe784b1a9a9e3db01d40b16ad04e7174dc6f7eb23440485ba06")
-    version(
-        "9.3.2", tag="v9.3.2", commit="5d2ddacb273682daa014ae22f17f3575e05b411e", deprecated=True
-    )
-    version(
-        "9.3.0", tag="v9.3.0", commit="9b8e9c1f16d4616b827d2d36955604a8e3fb915c", deprecated=True
-    )
-    version(
-        "9.2.0", tag="v9.2.0", commit="3a6ad66df7294417cf61618acdcfcc0fecccb045", deprecated=True
-    )
-    version(
-        "9.1.0", tag="v9.1.0", commit="df6d090061bae7ff2ba5a6bd57bb2ecbf538ef7a", deprecated=True
-    )
-    version(
-        "8.2.1", tag="v8.2.1", commit="939afcbad1a8273636a3686a31b51dae4f1f0c11", deprecated=True
-    )
 
-    depends_on("c", type="build")  # generated
-    depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
 
-    variant(
-        "openmp",
-        default=True,
-        description="Enable OpenMP support for ParseAPI " "(version 10.0.0 or later)",
-    )
+    variant("openmp", default=True, description="Enable OpenMP support for ParseAPI ")
 
     variant("static", default=False, description="Build static libraries")
 
     variant("stat_dysect", default=False, description="Patch for STAT's DySectAPI")
 
-    boost_libs = "+atomic+chrono+date_time+filesystem+system+thread+timer"
-    "+container+random+exception"
-
-    depends_on("boost@1.61.0:" + boost_libs, when="@10.1.0:")
-    depends_on("boost@1.61.0:1.69" + boost_libs, when="@:10.0")
-    depends_on("boost@1.67.0:" + boost_libs, when="@11.0.0:")
-    depends_on("boost@1.70.0:" + boost_libs, when="@12:12.3.0")
-    depends_on("boost@1.71.0:" + boost_libs, when="@13:")
+    depends_on(
+        "boost+atomic+chrono+date_time+filesystem+system+thread+timer+container+random+exception"
+    )
+    depends_on("boost@1.61.0:", when="@10.1.0:")
+    depends_on("boost@1.67.0:", when="@11.0.0:")
+    depends_on("boost@1.70.0:", when="@12:12.3.0")
+    depends_on("boost@1.71.0:", when="@13:")
 
     depends_on("libiberty+pic")
 
     # Dyninst uses elfutils starting with 9.3.0, and used libelf
     # before that.
     # NB: Parallel DWARF parsing in Dyninst 10.2.0 requires a thread-
     #     safe libdw
+    depends_on("elfutils", type="link")
```

Comment:
I'm curious. Is this needed given there are constrained versions?

Reply:
No, I just think it's more readable to say "it depends on elfutils", and then give further constraints on versions